cabal.project

```diff
@@ -29,18 +29,56 @@ allow-newer: windns-0.1.0.1:base
 constraints: rere -rere-cfg
 constraints: these -assoc
 
-program-options
-  -- So us hackers get all the assertion failures early:
-  --
-  -- NOTE: currently commented out, see
-  -- https://github.com/haskell/cabal/issues/3911
-  --
-  -- ghc-options: -fno-ignore-asserts
-  --
-  -- as a workaround we specify it for each package individually:
+-- So us hackers get all the assertion failures early:
+--
+-- NOTE: currently commented out, see
+-- https://github.com/haskell/cabal/issues/3911
+-- as a workaround we specify it for each package individually:
+--
+-- program-options
+--   ghc-options: -fno-ignore-asserts
+--
+-- NOTE: for library coverage in multi-project builds,
+-- see:
+--
+-- * https://github.com/haskell/cabal/issues/6440
+-- * https://github.com/haskell/cabal/issues/5213#issuecomment-586517129
+--
+-- We must mask coverage for dependencies of `cabal-install` in
+-- multiproject settings in order to generate coverage for
+-- the `cabal-install` library
```

Comment:
Question: how do we get the coverage for Cabal?

Reply:
Through pain and suffering. We'd have to rework the `cabal-testsuite`, and associated tests to be coverable. I'm punting on that one until i understand it better.